### ISSUE TITLE
Billy sessions do not function like Scrapelib's

### ISSUE THREAD

**Issue by mileswwatkins** [openstates/billy#251]
This session issue has hit both @rshorey and myself. I'll look into it on Monday. I believe it has to do with cookies and headers, but quick testing at HTTPBin hasn't turned up anything insightful.

```python
import lxml.html
import requests

from billy.scrape.legislators import LegislatorScraper
import scrapelib


class ALLegislatorScraper(LegislatorScraper):
    def _change_selected_session(self, session_object):
        DESIRED_SESSION_NAME = "Regular Session 1999"
        SESSION_URL = 'http://alisondb.legislature.state.al.us/Alison/ALISONLogin.aspx'

        # If cookies are working, ALISON should remember that the session changed
        form = {'__EVENTTARGET': 'ctl00$cboSession', 'ctl00$cboSession': DESIRED_SESSION_NAME}
        session_object.post(SESSION_URL, data=form, allow_redirects=False)

        # Report back the selected session
        doc = lxml.html.fromstring(session_object.get(SESSION_URL).text)
        selected_session = doc.xpath('//option[@selected]')[0].text_content()
        print(selected_session)

    def scrape(self, term, chambers):
        # Both print 1999; the legislative session is correctly changed
        self._change_selected_session(requests.Session())
        self._change_selected_session(scrapelib.Scraper())

        # Prints 2015; the legislative session change has is not remembered
        self._change_selected_session(self)
```

**Comment by mileswwatkins** [comment]
TLDR fix: the code above will run _correctly_ (ie, "1999" printed for all three `_change_selected_session` calls) if the `--fast` option is not used for billy.

---

billy's `Scraper` has the following lines:
```python
        if fastmode:
            self.requests_per_minute = 0
            self.cache_write_only = False
```

`cache_write_only` alters the scrapelib `CachingSession`, but I'm still trying to figure out how exactly it impedes cookies/headers from passing from one request to the next. For example, if I `print(session_object.cookies)` after both requests in `_change_selected_session`, all three sessions (`requests.Session`, `scrapelib.Scraper`, and `billy.scrape.legislators.LegislatorScraper`) show that the cookie is constant between calls (even with `--fast` on).

**Comment by mileswwatkins** [comment]
Well, at least for now we know how to make a session act sanely. I'm having difficulty pinning down _why_ this is myself, but will work on refactoring AL to use scrapeshell again.

**Comment by jamesturk** [comment]
since billy is basically dead letting this die too

**Comment by chishaku** [comment]
@jamesturk I've read a bit on the OpenStates blog but I'm still a bit confused about the future of this project.  Could you clarify your last comment about the future of this library?

Is all future development being moved to the opencivicdata/pupa library?

**Comment by jamesturk** [comment]
Yes, we're going to begin migrating scrapers to the Open Civic Data infrastructure.  pupa supercedes billy for scraping purposes, and new tools to replace some of billy's other features (billy.web.*) will be built in coming months
